### PR TITLE
add os_interfaces to compute

### PR DESCRIPTION
**Pull request by andypf**
@Carthaca: please take a look

### CODE REVIEW

File: lib/fog/compute/openstack/requests/create_os_interface.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  module Compute
+    class OpenStack
+      class Real
+        # === Parameters
+        # * server_id <~String>
+        # * options <~Hash>
+        def create_os_interface(server_id, options={})
+          body = {
+            'interfaceAttachment' => {
+              'net_id' => options[:net_id]
```
Comment (Carthaca) [inline lib/fog/compute/openstack/requests/create_os_interface.rb]: - `net_id`is optional
- and it is not allowed if `port_id` is specified

File: lib/fog/compute/openstack/models/os_interface.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'fog/openstack/models/model'
+
+module Fog
+  module Compute
+    class OpenStack
+      class OsInterface < Fog::OpenStack::Model
+        identity  :port_id
+        attribute :fixed_ips, type: :array
```
Comment (Carthaca) [inline lib/fog/compute/openstack/models/os_interface.rb]: hash rocket syntax pls, i.e. `:type => :array`

File: lib/fog/compute/openstack/models/os_interfaces.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+require 'fog/openstack/models/collection'
+require 'fog/compute/openstack/models/network'
```
Comment (Carthaca) [inline lib/fog/compute/openstack/models/os_interfaces.rb]: c&p error: `..os_interface`

File: lib/fog/compute/openstack/requests/delete_os_interface.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    class OpenStack
+      class Real
+        # === Parameters
+        # * server_id <~String>
+        # * port_id <~String>
+        def delete_os_interface(server_id, port_id)
+          request(
+            :expects => [202,204],
+            :method  => 'DELETE',
+            :path    => "/servers/#{server_id}/os-interface/#{port_id}"
```
Comment (Carthaca) [inline lib/fog/compute/openstack/requests/delete_os_interface.rb]: extra `/` at the beginning

**Review by Carthaca — CHANGES_REQUESTED** [review]
and I'm not sure about the `expects` i.e. the response status - api ref says otherwise, but we had occasions where it was good to allow more

**Comment by Carthaca** [comment]
and to get it upstream we would at least need the mock implementation and unit tests - even better vcr

File: lib/fog/compute/openstack.rb [hunk 5]
```diff
@@ -318,8 +318,23 @@ def self.data
                 'ram'                         => 51200
               },
               :volumes                   => {},
-              :snapshots                 => {}
-            }
+              :snapshots                 => {},
+              :os_interfaces             => {
+                [
+                  {
+                    "fixed_ips": [
```
Comment (Carthaca) [inline lib/fog/compute/openstack.rb]: and again the need for hash rockets to keep the ruby 1.9 support 🚀

**Review by andypf — COMMENTED** [review]
All changes have been implemented

**Pull request merged**